internal/report/report.go

```diff
@@ -92,9 +103,15 @@ func (r *Reporter) SendReport(ctx context.Context) (*DataResponse, error) {
 	rCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
 	defer cancel()
 
-	var m runtime.MemStats
-	runtime.ReadMemStats(&m)
-	r.body["heap_usage_bytes"] = strconv.FormatUint(m.Alloc, 10)
+	r.gatherersMtx.Lock()
+	for key, g := range r.gatherers {
+		var err error
+		r.body[key], err = g(rCtx)
+		if err != nil {
+			return nil, fmt.Errorf("gather telemetry error for key %s: %w", key, err)
+		}
+	}
+	r.gatherersMtx.Unlock()
 
 	resp, err := r.client.WithJSON(r.body).Do(rCtx, "POST", "/v1/version")
```

Comment:
We could have gatherers. But `r.body` being internal I don't see a race for it.

Reply:
It depends on under what conditions `SendReport` is called. If it's ever only called sequentially, we don't have a problem; if it can be called concurrently, then e.g. a slow (extremely) HTTP call, or if something locks up during gatherer registration, could have another go-rutine come in and modify `r.body` while in transit.

Reply:
In this case only one routine calls `SendReport`.